reactivesocket-client/src/main/java/io/reactivesocket/client/LoadBalancer.java

```diff
@@ -618,4 +661,284 @@ public void shutdown() {}
         @Override
         public void close() throws Exception {}
     }
+
+    /**
+     * Wrapper of a ReactiveSocket, it computes statistics about the req/resp calls and
+     * update availability accordingly.
+     */
+    private class WeightedSocket extends ReactiveSocketProxy {
+        private static final double STARTUP_PENALTY = Long.MAX_VALUE >> 12;
+
+        private final ReactiveSocket child;
+        private ReactiveSocketFactory<?> factory;
+        private final Quantile lowerQuantile;
+        private final Quantile higherQuantile;
+        private final long inactivityFactor;
+
+        private volatile int pending;       // instantaneous rate
```

Comment:
I'm not focusing on performance here, but that's a micro-optimization that barely impact the readability of the code.
So I would like to keep it.